### ISSUE TITLE
[10.0] TransactionRollbackError while deleting outgoing email

### ISSUE THREAD

**Issue by eLBati** [odoo/odoo#22148]
**Impacted versions:**
[10.0]

**Steps to reproduce:**
I am not able to reproduce it systematically, it happens sometimes while sending emails by SMTP server.
See the error below.
The effect is the email is sent twice.
Maybe the error message can help to understand the problem or somebody else already got into this...

```
2018-01-10 10:00:27,250 15436 INFO agilebg odoo.addons.mail.models.mail_mail: Mail with ID 87686 and Message-Id u'<1515578414.252722978591919.143793211405891-openerp-12031-project.issue@odoo10.agilebg.com>' successfully sent
2018-01-10 10:00:28,258 15436 INFO agilebg odoo.sql_db: bad query: DELETE FROM mail_mail WHERE id IN (87686)
2018-01-10 10:00:28,280 15436 INFO agilebg werkzeug: 127.0.0.1 - - [10/Jan/2018 10:00:28] "POST /web/dataset/call_kw/project.issue/message_post HTTP/1.0" 500 - 
2018-01-10 10:00:28,814 15436 ERROR agilebg werkzeug: Error on request:
Traceback (most recent call last):
  File "/srv/odoo-10-agilebg-prod/agilebg/odoo/lib/python2.7/site-packages/werkzeug/serving.py", line 193, in run_wsgi
    execute(self.server.app)
  File "/srv/odoo-10-agilebg-prod/agilebg/odoo/lib/python2.7/site-packages/werkzeug/serving.py", line 181, in execute 
    application_iter = app(environ, start_response)
  File "/srv/odoo-10-agilebg-prod/agilebg/odoo/lib/python2.7/site-packages/odoo/service/wsgi_server.py", line 186, in application
    return application_unproxied(environ, start_response)
  File "/srv/odoo-10-agilebg-prod/agilebg/odoo/lib/python2.7/site-packages/odoo/service/wsgi_server.py", line 172, in application_unproxied
    result = handler(environ, start_response)
  File "/srv/odoo-10-agilebg-prod/agilebg/odoo/lib/python2.7/site-packages/odoo/http.py", line 1308, in __call__
    return self.dispatch(environ, start_response)
  File "/srv/odoo-10-agilebg-prod/agilebg/odoo/lib/python2.7/site-packages/odoo/http.py", line 1282, in __call__
    return self.app(environ, start_wrapped)
  File "/srv/odoo-10-agilebg-prod/agilebg/odoo/lib/python2.7/site-packages/werkzeug/wsgi.py", line 599, in __call__
    return self.app(environ, start_response)
  File "/srv/odoo-10-agilebg-prod/agilebg/odoo/lib/python2.7/site-packages/odoo/http.py", line 1486, in dispatch
    response = self.get_response(httprequest, result, explicit_session)
  File "/srv/odoo-10-agilebg-prod/agilebg/odoo/lib/python2.7/site-packages/odoo/http.py", line 278, in __exit__
    self._cr.commit()
  File "/srv/odoo-10-agilebg-prod/agilebg/odoo/lib/python2.7/site-packages/odoo/sql_db.py", line 154, in wrapper
    return f(self, *args, **kwargs)
  File "/srv/odoo-10-agilebg-prod/agilebg/odoo/lib/python2.7/site-packages/odoo/sql_db.py", line 378, in commit
    func()
  File "/srv/odoo-10-agilebg-prod/agilebg/odoo/lib/python2.7/site-packages/odoo/addons/mail/models/res_partner.py", line 248, in send_notifications
    env['mail.mail'].browse(email_ids).send()
  File "/srv/odoo-10-agilebg-prod/agilebg/odoo/lib/python2.7/site-packages/odoo/addons/mail/models/mail_mail.py", line 297, in send
    mail._postprocess_sent_message(mail_sent=mail_sent)
  File "/srv/odoo-10-agilebg-prod/agilebg/odoo/lib/python2.7/site-packages/odoo/addons/portal_sale/models/mail_mail.py", line 23, in _postprocess_sent_message
    return super(Mail, self)._postprocess_sent_message(mail_sent=mail_sent)
  File "/srv/odoo-10-agilebg-prod/agilebg/odoo/lib/python2.7/site-packages/odoo/addons/mass_mailing/models/mail_mail.py", line 107, in _postprocess_sent_message
    return super(MailMail, self)._postprocess_sent_message(mail_sent=mail_sent)
  File "/srv/odoo-10-agilebg-prod/agilebg/odoo/lib/python2.7/site-packages/odoo/addons/mail/models/mail_mail.py", line 148, in _postprocess_sent_message
    self.sudo().filtered(lambda self: self.auto_delete).unlink()
  File "/srv/odoo-10-agilebg-prod/agilebg/odoo/lib/python2.7/site-packages/odoo/addons/mail/models/mail_mail.py", line 71, in unlink
    res = super(MailMail, self).unlink()
  File "/srv/odoo-10-agilebg-prod/agilebg/odoo/lib/python2.7/site-packages/odoo/models.py", line 3405, in unlink
    cr.execute(query, (sub_ids,))
  File "/srv/odoo-10-agilebg-prod/agilebg/odoo/lib/python2.7/site-packages/odoo/sql_db.py", line 154, in wrapper
    return f(self, *args, **kwargs)
  File "/srv/odoo-10-agilebg-prod/agilebg/odoo/lib/python2.7/site-packages/odoo/sql_db.py", line 231, in execute
    res = self._obj.execute(query, params)
TransactionRollbackError: deadlock detected
DETAIL:  Process 15480 waits for ShareLock on transaction 6319240; blocked by process 16095.
Process 16095 waits for ShareLock on transaction 6319234; blocked by process 15480.
HINT:  See server log for query details.
CONTEXT:  while deleting tuple (4,11) in relation "mail_mail"
```

**Expected behavior:**
No error message and the email is sent once

**Comment by eLBati** [comment]
I made an OCA module
https://github.com/OCA/social/pull/227
that should also mitigate this

**Comment by Yenthe666** [comment]
Dear @eLBati,

As Odoo 13 has been released last week it means that our support for Odoo version 10 has been stopped.
If you can still reproduce this issue on a newer version (version 11, 12 or 13 at the point of writing) then please do respond back or create a new issue.
Sorry that we didn't get around to resolving your issue.

Regards,
Yenthe
